src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -463,6 +470,315 @@ int RGWBucket::check_object_index(const DoutPrefixProvider *dpp,
   return 0;
 }
 
+int RGWBucket::check_index_olh(const DoutPrefixProvider *dpp, RGWBucketAdminOpState& op_state,
+                                     RGWFormatterFlusher& flusher, const int shard, 
+                                     uint64_t* const count_out, optional_yield y) {
+  string marker = BI_OLH_ENTRY_NS_START;
+  bool is_truncated = true;
+  list<rgw_cls_bi_entry> entries;
+
+  rgw::sal::RadosStore* rados_store = dynamic_cast<rgw::sal::RadosStore*>(driver);
+  if (!rados_store) {
+    return 0;
+  }
+  RGWRados* store = rados_store->getRados();
+  RGWObjectCtx obj_ctx(driver);
+
+  RGWRados::BucketShard bs(store);
+  int ret = bs.init(dpp, bucket->get_info(), get_bucket_info().layout.current_index, shard, y);
+  if (ret < 0) {
+    ldpp_dout(dpp, -1) << "ERROR bs.init(bucket=" << bucket << "): " << cpp_strerror(-ret) << dendl;
+    return ret;
+  }
+  
+  *count_out = 0;
+  do {
+    entries.clear();
+    ret = store->bi_list(bs, "", marker, -1, &entries, &is_truncated, y);
+    if (ret < 0) {
+      ldpp_dout(dpp, -1) << "ERROR bi_list(): " << cpp_strerror(-ret) << dendl;
+      break;
+    }
+    list<rgw_cls_bi_entry>::iterator iter;
+    for (iter = entries.begin(); iter != entries.end(); ++iter) {
+      rgw_cls_bi_entry& entry = *iter;
+      marker = entry.idx;
+      if (entry.type != BIIndexType::OLH) {
+        is_truncated = false;
+        break;
+      }
+      rgw_bucket_olh_entry olh_entry;
+      auto iiter = entry.data.cbegin();
+      try {
+        decode(olh_entry, iiter);
+      } catch (buffer::error& err) {
+        ldpp_dout(dpp, -1) << "ERROR failed to decode olh entry for key: " << entry.idx << dendl;
+        continue;
+      }
+      if (!olh_entry.exists && olh_entry.pending_removal) {
```

Comment:
this if-block appears to cover the rest of this for loop. we could save some nesting below by inverting the condition and `continue`ing instead. i think that would make it easier to follow the ifs and if-elses below